Rules/Warns/Zombie_WarnsCommon.as

```diff
@@ -0,0 +1,206 @@
+// SonantDread @ October 11th 2024
+#define SERVER_ONLY
+#include "Zombie_GlobalMessagesCommon.as";
+
+u8 maxWarns = 3; // max warns before ban
+u16 warnDuration = 30; // days
+u16 banTime = 30; // days
+
+const string fileName = "Zombie_Warns.cfg";
+
+void onInit(CRules@ this)
+{
+    ConfigFile cfg = ConfigFile();
+    if (cfg.loadFile(fileName))
+    {
+        maxWarns = cfg.exists("max_warns") ? cfg.read_u8("max_warns") : 3;
+        warnDuration = cfg.exists("warn_duration") ? cfg.read_u16("warn_duration") : 30;
+        banTime = cfg.exists("ban_time") ? cfg.read_u16("ban_time") : 30;
+    }
+}
+
+void WarnPlayer(CPlayer@ admin, string playerName, u32 duration, string reason)
+{
+    string player = getPlayerUsername(playerName);
+    string adminUsername = admin.getUsername();
+    ConfigFile@ cfg = getWarnsConfig();
+
+    // ignore if player is admin or the person who is warning
+    CPlayer@ playerObject = getPlayerByUsername(player);
+    if (adminUsername == player || (playerObject !is null && !playerObject.isMod())) return;
+
+    array<string> reasons;
```

Comment:
i understand that ``array<type>`` is the standard angelscript array syntax, but for kag purposes please use ``type[]`` instead.